From d5481abfea96a1ee192e879c9f6805d042cd85a4 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Thu, 01 Oct 2020 18:30:58 +0300
Subject: [PATCH] [SW-20246] gaudi: update security bit in status

Update security device status bit to true if
security is enabled, or to false if otherwise.

Change-Id: I2c3e7307248a33f86ebc2b5b98819a3e10dba6dd
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 8c3e987..3e0f451 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -954,15 +954,19 @@
 
 void preboot_security_init(void)
 {
+	bool is_sec_en;
+
 	cpu_boot_err_clrset(CPU_BOOT_ERR0_SECURITY_NOT_RDY, true);
 	preboot_puts("Security init started...\n");
 
-	if (preboot_security_is_enabled()) {
+	is_sec_en = preboot_security_is_enabled();
+	if (is_sec_en) {
 		preboot_puts("Security is enabled\n");
 	} else {
 		preboot_puts("Security is disabled\n");
 		/* TODO: add preboot_security_pci_disable(); */
 	}
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_SECURITY_EN, is_sec_en);
 
 	/*
 	 * TODO: Remove this call once it's added to the security enable check
